idl/internal/lex.rl

```diff
@@ -71,17 +74,28 @@ func (lex *lexer) Lex(out *yySymType) int {
                 lex.linesSinceDocstring = 0
             };
 
-        ws = [ \t\r];
+        ws = [ \r];
+
+        # Tabs are a single byte in the data buffer but should account for some
+        # additional amount of whitespace for our column calculations. This is
+        # computed based on the current "virtual column" (taking into account
+        # any prior tab expansion on this line) and the configured tab width.
+        # The result is the amount of *additional* column space added by this
+        # tab (beyond the single byte contributed by the tab character itself).
+        tab = '\t' >{
+            lex.tabExtra += lex.tabWidth - (lex.te - lex.lineStart + lex.tabExtra) % lex.tabWidth - 1
+        };
 
         # All uses of \n MUST use this instead if we want accurate line
-        # number tracking.
+        # and column tracking.
         newline = '\n' >{
             lex.line++
-            lex.lineStart = lex.p + 1
+            lex.lineStart = lex.te + 1
```

Comment:
This doesn't change anything in practice, but it's slightly more accurate to refer to the "end of the newline character" here rather than the current "pointer" in the data buffer.